.github/workflows/clang-tidy.yaml

```diff
@@ -0,0 +1,26 @@
+name: clang-tidy
+
+on: [pull_request]
+
+jobs:
+  job:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install GoogleTest.
+        run: sudo apt install -y libgtest-dev
+
+      - name: Generate build files and compilation database.
+        run: cmake -B out
+
+      - name: Lint with clang-tidy.
+        uses: ZedThree/clang-tidy-review@v0.7.1
+        id: review
+        with:
+          build_dir: out
```

Comment:
Recently clang-tidy-review supports specifying a `.clang-tidy` config file, and its default location is empty. I think we need to use this feature to make sure it uses our custom config file.
https://github.com/ZedThree/clang-tidy-review/commit/0ad0fe152ac38306c2a987a861593354208bb6ab